### PR TITLE
Performance enhancement to SSI's ssi_recentTopics

### PR DESCRIPTION
**Pull request by live627**
Use two faster queries instead of that slow monstrosity.

### CODE REVIEW

File: SSI.php [hunk 1]
```diff
@@ -451,35 +451,54 @@ function ssi_recentTopics($num_recent = 8, $exclude_boards = null, $include_boar
...
 			'include_boards' => empty($include_boards) ? '' : $include_boards,
 			'exclude_boards' => empty($exclude_boards) ? '' : $exclude_boards,
 			'min_message_id' => $modSettings['maxMsgID'] - 35 * min($num_recent, 5),
 			'is_approved' => 1,
 		)
 	);
+	$topics = array();
+	while ($row = wesql::fetch_assoc($request))
+		$topics[$row['id_topic']] = $row;
+	wesql::free_result($request);
```
Comment (emanuele45) [inline SSI.php:476]: /me feels there is something wrong here. :P

**Comment by Arantor** [comment]
o.O I didn't study it that closely, I just looked over the changes to the query as a whole which to me was the more important part :P

**Comment by Arantor** [comment]
And fixed.

File: SSI.php (hunk 1 above)
Comment (live627) [inline SSI.php:476]: Major oops.

**Pull request merged**